geomstats/learning/polynomial_regression.py

```diff
@@ -0,0 +1,572 @@
+        Optional, default: 0.1
+    tol : float
+        Tolerance for loss minimization.
+        Optional, default: 1e-5
+    verbose : bool
+        Verbose option.
+        Optional, default: False.
+    initialization : str or array-like,
+        {'random', 'data', 'frechet', warm_start'}
+        Initial values of the parameters for the optimization,
+        or initialization method.
+        Optional, default: 'random'
+    regularization : float
+        Weight on the constraint for the intercept to lie on the manifold in
+        the extrinsic optimization scheme. An L^2 constraint is applied.
+        Optional, default: 1.
+    """
+
+    def __init__(
+        self,
+        space,
+        order,
+        metric=None,
+        center_X=True,
+        method="extrinsic",
+        max_iter=100,
+        init_step_size=0.1,
+        tol=1e-5,
+        verbose=False,
+        initialization="random",
+        regularization=1.0,
+    ):
+        if metric is None:
+            metric = space.metric
+        self.metric = metric
+        self.space = space
+
+        # For now, only allow for fitting through quintic polynomials
+        geomstats.errors.check_parameter_accepted_values(
+            order, "order", [i + 1 for i in range(0, 5)]
+        )
+        self.order = order
+
+        self.intercept_ = None
+        self.coef_ = None
+        self.center_X = center_X
+        self.mean_ = None
+        self.training_score_ = None
+        geomstats.errors.check_parameter_accepted_values(
+            method, "method", ["extrinsic", "riemannian"]
+        )
+        self.method = method
+        self.max_iter = max_iter
+        self.verbose = verbose
+        self.init_step_size = init_step_size
+        self.tol = tol
+        self.initialization = initialization
+        self.regularization = regularization
+
+    def _model(self, X, coef, intercept):
+        """Compute the generative model of the polynomial regression.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape=[...,}]
+            Training input samples.
+        coef : array-like, shape=[..., {dim, [n,n]}]
+            Coefficient of the polynomial regression.
+        intercept : array-like, shape=[..., {dim, [n,n]}]
+            Intercept of the polynomial regression.
+
+        Returns
+        -------
+        _ : array-like, shape=[..., {dim, [n,n]}]
+            Value on the manifold output by the generative model.
+        """
+        X_copy = X[:, None]
+
+        # Pads additional dimensions as needed
+        # generate exponentiated matrix - N (samples) x k (order)
+        # This is a matrix where columns are powers of X
+        X_powers = gs.hstack([X_copy**k for k in range(1, self.order + 1)])
+
+        # Coef is of shape k x dim (where dim can be multidimensional)
+        # Output should be shape N x dim -> exp -> N
+        # Matrix multiply with coefficients and apply exp map at intercept
```

Comment:
This comment is not helpful because it is redundant with the code that follows: if the code is well-written enough, then that comment is not needed. 

The code is good -> remove that comment.